tests/acceptance/api_test.go

```diff
@@ -2415,8 +2525,14 @@ func createAdminRole(t *testing.T, awsSession client.ConfigProvider, adminRoleNa
 		})
 		require.Nil(t, err)
 	}
-	// IAM Role takes a while to propagate....
-	//time.Sleep(10 * time.Second)
+
+	// Wait for the role to be assumable
+	log.Println("Created admin test role. Waiting for role to be assumeable")
+	testutil.Retry(t, 30, time.Millisecond*500, func(r *testutil.R) {
```

Comment:
Turns out a lot of our `apiRequest` retries were only necessary because our IAM role wasn't yet ready to use.
Waiting for this to be ready upfront reduces some noise in tests.